### DISCUSSION TITLE
Heavy memory use

### DISCUSSION THREAD

**Discussion by msklywenn** [outline/outline#7714] [Self Hosting] [answered]
Hello,

I've noticed our server's memory use as raised a lot since approximately mid July to the point of over-commiting which I find a bit worrying. From what I could gather, most of the use comes from Outline. After restarting it, I have a nodejs process that shows 11.3GB in htop's virtual memory column and a second one showing 11.4GB (are they sharing the same virtual space? I don't know). Before restarting it, I believe that one was at a whopping 21GB.

![image](https://github.com/user-attachments/assets/1d448fa5-8b07-4ab9-8604-0a34ac4966d3)

Our use of Outline hasn't increased that much, in fact, it probably lowered quite a lot since May, so I don't think we are doing anything crazy with it. We have no more than 10 users, only 2 actually actively writing.

I usually wait a bit before upgrading outline. But I think I upgraded quite fast to 0.78.0, since we were very happy to use the new features from that version! So I think the big memory increase came with that version.

Is there a way to limit outline's memory use? Could someone investigate outline's general memory allocations to see if there is anything to be done?

All that said, I'd also like to thanks the outline team for such a great piece of software.

**Comment by tommoor** [comment, marked as answer]
I think you should measure Outline's memory use more directly through `ps` or similar, our production servers run on 1gb of available memory – which granted I've been thinking this should be measured and reduced, but certainly nothing like 12gb+

**Comment by msklywenn** [comment]
Indeed, I looked at virtual memory use when I should have looked at resident memory use. Resident memory use is around 250MB so it's more than fine. I'd like to know why virtual memory use is that high though.

For my problem, I'm going to check resident memory use of other apps. I'll check back outline in a few weeks though, maybe it used much more resident memory when it was using 21GB of virtual memory.
